src/rust/lib/enso-prelude/src/result.rs

```diff
@@ -0,0 +1,19 @@
+//! This module defines utilities for working with the `Result` type.
+
+/// Adds utilities to the `Result` type.
+pub trait ResultOps {
+    type Item;
+    type Error;
+
+    /// Call the given handler if this is an error and promote `Result` to `Option`.
+    fn handle_err<F>(self, f:F) -> Option<Self::Item> where F : FnOnce(Self::Error);
```

Comment:
This function is cool! As I remember, it could be used in many places in `node_editor.rs` file. If you want you can look through all `if let Err(` and replace them with this.

Reply:
I don't think this really fits there, as these `if let Err`s never involve promotion to the option after error is handled.
If we just want to do something if there's an error, I consider `if let Err` idiomatic and reasonable way.